bindings/rust/bench/src/s2n_tls.rs

```diff
@@ -0,0 +1,207 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    harness::{read_to_bytes, Mode, TlsBenchHarness},
+    CA_CERT_PATH, SERVER_CERT_CHAIN_PATH, SERVER_KEY_PATH,
+};
+use log::debug;
+use s2n_tls::{
+    callbacks::VerifyHostNameCallback,
+    config::{Builder, Config},
+    connection::Connection,
+    enums::Blinding,
+    security::DEFAULT_TLS13,
+};
+use std::{
+    cell::UnsafeCell,
+    collections::VecDeque,
+    ffi::c_void,
+    io::{Read, Write},
+    os::raw::c_int,
+    pin::Pin,
+    task::Poll::Ready,
+};
+
+pub struct SignalToNoise {
+    // UnsafeCell is needed b/c client and server share *mut to IO buffers
+    // Pin<Box<T>> is to ensure long-term *mut to IO buffers remain valid
+    client_to_server_buf: Pin<Box<UnsafeCell<VecDeque<u8>>>>,
+    server_to_client_buf: Pin<Box<UnsafeCell<VecDeque<u8>>>>,
+    client_config: Config,
+    server_config: Config,
+    client_conn: Connection,
+    server_conn: Connection,
+    client_handshake_completed: bool,
+    server_handshake_completed: bool,
+}
+
+/// Custom callback for verifying hostnames, need it to use s2n-tls safely
```

Comment:
Nit: 🤔 Sort of confused about how you phrased this; is this callback really for using s2n-tls safely? I don't know if we're worried about safety during benchmarking. What would be the impact if we didn't implement this callback?

Reply:
We would still have to implement the callback, but it would just return true and probably optimize out. Rustls requires you to give it a hostname, so it runs the hostname verification; implementing it for s2n-tls makes a more fair comparison.